File: src/fiefapp.svelte/src/models/industries/subsidiary.ts
import { Industry } from './industry';

interface ISubsidiary {
  quality: number;
  developmentLevel: number;
  daysworkThisYear: number;
  incomeFactor: number;
  silverPortion: number;
  basePortion: number;
  luxuryPortion: number;
  daysworkBuild: number;
  daysworkUpkeep: number;
  springModifier: number;
  summerModifier: number;
  fallModifier: number;
  winterModifier: number;
}

export class Subsidiary extends Industry implements ISubsidiary {
  daysworkThisYear: number;
  name: string;
  quality: number;
  developmentLevel: number;
  incomeFactor: number;
  silverPortion: number;
  basePortion: number;
  luxuryPortion: number;
  daysworkBuild: number;
  daysworkUpkeep: number;
  springModifier: number;
  summerModifier: number;
  fallModifier: number;
  winterModifier: number;

  constructor() {
    super();

    this.name = '';
    this.quality = 0;
    this.developmentLevel = 0;
    this.incomeFactor = 0;
    this.silverPortion = 0;
    this.basePortion = 0;
    this.luxuryPortion = 0;
    this.daysworkBuild = 0;
    this.daysworkUpkeep = 0;
    this.springModifier = 0;
    this.summerModifier = 0;
    this.fallModifier = 0;
    this.winterModifier = 0;
    this.needSteward = true;
    this.daysworkThisYear = 0;
    this.silver = 0;
    this.base = 0;
    this.luxury = 0;
  }
}

File: src/fiefapp.entities/Subsidiary.cs
using fiefapp.entities.BaseClasses;
using fiefapp.entities.Interfaces;
using System;

namespace fiefapp.entities
{
    public class Subsidiary : Industry, IBaseEntity, IAffectedByWeather, IFiefForeinKey
    {
        public override Guid Id { get; set; }
        public Guid FiefId { get; set; }
        public override string Name { get; set; }
        public override int? Silver { get; set; }
        public override int? Base { get; set; }
        public override int? Luxury { get; set; }
        public override int? Wood { get; set; }
        public override int? Stone { get; set; }
        public override int? Iron { get; set; }
        public override int? Quality { get; set; }
        public override int DevelopmentLevel { get; set; }
        public int DaysworkThisYear { get; set; }
        public int DaysworkUpkeep { get; set; }
        public int DaysworkBuild { get; set; }
        public int IncomeFactor { get; set; }
        public decimal SilverPortion { get; set; }
        public decimal BasePortion { get; set; }
        public decimal LuxuryPortion { get; set; }
        public decimal SpringModifier { get; set; }
        public decimal SummerModifier { get; set; }
        public decimal FallModifier { get; set; }
        public decimal WinterModifier { get; set; }
        public override bool IsBeingDeveloped { get; set; }
        public override bool IsInformationItem { get; set; }
        public override Guid? StewardId { get; set; }
        public override Steward Steward { get; set; }
        public override Guid? DevelopmentId { get; set; }
        public override Development Development { get; set; }
    }
}
